tests/layer_validation_tests.cpp

```
@@ -28254,6 +28254,48 @@ TEST_F(VkPositiveLayerTest, MultiplaneImageTests) {
     vkDestroyImage(device(), image, NULL);
 }
 
+TEST_F(VkPositiveLayerTest, ApiVersionZero) {
+    TEST_DESCRIPTION("Check that apiVersion = 0 is valid.");
+    m_errorMonitor->ExpectSuccess();
+    app_info.apiVersion = 0U;
+    ASSERT_NO_FATAL_FAILURE(InitFramework(myDbgFunc, m_errorMonitor));
+    m_errorMonitor->VerifyNotFound();
+}
+
+// The tests rely on desktop 1.1 Loader behavior that prevent illegal apiVersion settings from reaching a 1.0.x driver
+#ifndef ANDROID
+TEST_F(VkLayerTest, ApiVersionWarning) {
+    TEST_DESCRIPTION("Check that apiVersion > known versions is warned.");
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_WARNING_BIT_EXT,
+                                         "Unrecognized CreateInstance->pCreateInfo->pApplicationInfo.apiVersion number");
+    // A side effect of the intended warning is these other warnings
+    std::vector<const char *> expected = {
+        {"loader_add_to_layer_list: Explicit layer VK_LAYER_GOOGLE_threading is using an old API version",
+         "loader_add_to_layer_list: Explicit layer VK_LAYER_GOOGLE_unique_objects is using an old API version",
+         "loader_add_to_layer_list: Explicit layer VK_LAYER_LUNARG_core_validation is using an old API version",
+         "loader_add_to_layer_list: Explicit layer VK_LAYER_LUNARG_device_profile_api is using an old API version",
+         "loader_add_to_layer_list: Explicit layer VK_LAYER_LUNARG_object_tracker is using an old API version",
+         "loader_add_to_layer_list: Explicit layer VK_LAYER_LUNARG_parameter_validation is using an old API version"}};
+    for (auto warning : expected) {
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_WARNING_BIT_EXT, warning);
+    }
+
+    app_info.apiVersion = VK_MAKE_VERSION(999U, 999U, 999U);
+    ASSERT_NO_FATAL_FAILURE(InitFramework(myDbgFunc, m_errorMonitor));
```

@cnorthrop @ianelliottus  -- this line was failing in our Android CI environment... would skipping this test (or expecting failure) if  vkEnumerateInstanceVersion (on Android) returning 1.0 be sufficient to avoid an unexpected failure?

---

Confirmed this test is failing here even on more recent Android versions.  Should this be a loader test instead of layer, given that it is driven by loader functionality?

---

Hmm.  I should think about this some more.  This is an area where 1.1 changed semantics, which is unfortunate.  

Here's a quick thought.  There is a way of testing what version of loader you have (i.e. 1.0.* vs. 1.1 and beyond).  This is done by calling GIPA of the new vkEnumerateInstanceVersion global function.  If NULL, you have a 1.0 loader (which should return an error for >1.0.*).  If non-NULL, call the function and observe what version the loader supports (i.e. loader version == instance version).  You can hide this check around that logic.